entrypoint.sh

```diff
@@ -17,30 +17,81 @@ fi
 CMD="appium --log-no-colors --log-timestamp -pa /wd/hub --port $APPIUM_PORT --log $TASK_LOG --log-level $LOG_LEVEL $APPIUM_CLI $plugins_cli"
 #--use-plugins=relaxed-caps
 
+stop_video() {
+  local artifactId=$1
+  if [ -z ${artifactId} ]; then
+    echo "[warn] [Stop Video] artifactId param is empty!"
+    return 0
+  fi
+
+  if [ -f /tmp/${artifactId}.mp4 ]; then
+    ls -lah /tmp/${artifactId}.mp4
+    ffmpeg_pid=$(pgrep --full ffmpeg.*${artifactId}.mp4)
+    echo "[info] [Stop Video] ffmpeg_pid=$ffmpeg_pid"
+    kill -2 $ffmpeg_pid
+    echo "[info] [Stop Video] kill output: $?"
+
+    # wait until ffmpeg finished normally and file size is greater 48 byte! Time limit is 5 sec
+    idleTimeout=30
+    startTime=$(date +%s)
+    while [ $((startTime + idleTimeout)) -gt "$(date +%s)" ]; do
+      echo "[info] [Stop Video] videoFileSize: $(wc -c /tmp/${artifactId}.mp4 | awk '{print $1}') bytes."
+      echo -e "[info] [Stop Video] \n Running ffmpeg processes:\n $(pgrep --list-full --full ffmpeg) \n-------------------------"
+
+      if ps -p $ffmpeg_pid > /dev/null 2>&1; then
+        echo "[info] [Stop Video] ffmpeg not finished yet"
+        sleep 0.5
```

Comment:
0.5 seems too big :) how about 0.1 or 0.3? earlier 0.3 was some kind of magic pause which guarantee ~90% of valid mp4